# Test generated by RoostGPT for test credential-wallet-rest-assured using AI Type Azure Open AI and AI Model roost-gpt4-32k


# RoostTestHash=819791060f

config:
  target: "{{ $processEnvironment.API_HOST }}"
  phases:
    - duration: 10
      arrivalRate: 1
  payload:
    path: "persons_id_delete_others.csv"
    fields:
      - "id"
  plugins:
    expect: {}

scenarios:
  - name: "Non-success API Response Scenarios"
    flow:
      - log: 'Attempting to access API with invalid URL'
        get:
          url: '/invalid/url'
        expect:
          - statusCode: 404  
        think: 1
      - log: 'Attempting to DELETE person with unauthenticated access'
        delete:
          url: '/recipients/persons/{{ id }}'
        expect:
          - statusCode: 401  # Unauthorized access
          - hasProperty: 'error'
        think: 1
      - log: 'Attempting to DELETE person with authenticated access but incorrect ID'
        delete:
          url: '/recipients/persons/non_existent_id'
          headers:
            Token: "{{$processEnvironment.AUTH_TOKEN}}"
        expect:
          - statusCode: 400  # Bad Request
          - hasProperty: 'error'
        think: 1
